dts/bindings/usb/zephyr,hid-device.yaml

```diff
@@ -0,0 +1,50 @@
+# Copyright (c) 2023 Nordic Semiconductor ASA
+# SPDX-License-Identifier: Apache-2.0
+
+description: Bindings for virtual HID device
+
+compatible: "zephyr,hid-device"
+
+include: base.yaml
+
+properties:
+  interface-name:
+    type: string
+    description: |
+      HID device name. When this property is present, a USB device will use it
+      as the string descriptor of the interface.
+
+  boot-protocol-code:
+    type: string
+    required: true
+    description: |
+      Device can be used as a Boot Device and supports a predefined protocol
+      for mouse or keyboards.
+      - none:     Device does not support the boot interface
+      - keyboard: Device supports boot interface and keyboard protocol
+      - mouse:    Device supports boot interface and mouse protocol
+    enum:
+      - none
+      - keyboard
+      - mouse
+
+  polling-rate:
+    type: int
+    required: true
+    description: |
+      Input type reports polling rate in number of 125µs periods.
+      Depending on the actual speed, the calculated polling rate could be clamped
+      to 1ms or 255ms.
```

Comment:
The 125µs period is the case only on high-speed. On full-speed the meaning is different.

Reply:
I don't think it's exactly that either... On High Speed it's the formula of 2^(bInterval-1) = x (125ms). Where it would follow this table. Only Values of 1 to 16 are valid.
| bInterval | uS      |
|-----------|---------|
| 1         | 125     |
| 2         | 250     |
| 3         | 500     |
| 4         | 1000    |
| 5         | 2000    |
| 6         | 4000    |
| 7         | 8000    |
| 8         | 16000   |
| 9         | 32000   |
| 10        | 64000   |
| 11        | 128000  |
| 12        | 256000  |
| 13        | 512000  |
| 14        | 1024000 |
| 15        | 2048000 |
| 16        | 4096000 |

For Full/Low Speed it's, just 1ms units.

Reply:
Thanks for the point. I want to use 125µs granularity for this property. But the actual code does not handle it properly.